src/functions.jl

```diff
@@ -1,29 +1,29 @@
 @doc "Create a function call from a list of arguments"->
-function lang(f::SEXPREC,args...;kwargs...)
+function rlang_p(f, args...; kwargs...)
     argn = length(args)+length(kwargs)
-    l = preserve(sexp(ccall((:Rf_allocVector,libR),Ptr{Void},(Cint,Int),6,argn+1)))
-    s = l.p
-    ccall((:SETCAR,libR),Ptr{Void},(Ptr{Void},Ptr{Void}),s,f)
+    s = l = protect(allocArray(LangSxp,argn+1))
+    setcar!(s,sexp(f))
     for argv in args
-        s = ccall((:CDR,libR),Ptr{Void},(Ptr{Void},),s)
-        ccall((:SETCAR,libR),Ptr{Void},(Ptr{Void},Ptr{Void}),s,sexp(argv))
+        s = cdr(s)
+        setcar!(s,sexp(argv))
```

Comment:
From what I understand, I think it's fine as we're immediately binding it to a protected list. See, for example, the last block in [R-exts §5.11](http://cran.r-project.org/doc/manuals/r-patched/R-exts.html#Evaluating-R-expressions-from-C).